src/overlays/actors/ovl_En_Dns/z_en_dns.c

```diff
@@ -21,63 +56,466 @@ const ActorInit En_Dns_InitVars = {
+            case 0: // OK
+                switch (this->dnsItemEntry->purchaseableCheck(this)) {
+                    case 0:
+                        func_8010B720(globalCtx, 0x10A5);
+                        // "[sound 3882]Not enough Rupees!  Come back again!"
+                        this->actionFunc = func_809F008C;
+                        return;
+                    case 1:
+                        func_8010B720(globalCtx, 0x10A6);
+                        // "[sound 3882]You can't have this now! Come back again!"
+                        this->actionFunc = func_809F008C;
+                        return;
+                    case 3:
+                        func_8010B720(globalCtx, 0x10DE);
+                        // "[sound 3882]So sorry! You can't buy that right now!"
+                        this->actionFunc = func_809F008C;
+                        return;
+                    case 2:
+                    case 4:
+                        func_8010B720(globalCtx, 0x10A7);
+                        // "[sound 3880]Thank you very much!"
+                        this->actionFunc = func_809EFEE8;
+                        return;
+                }
+                return;
+            case 1: // No way
+                func_8010B720(globalCtx, 0x10A4);
+                // "[sound 3882]YIKES!! I'm going home then!"
+                this->actionFunc = func_809F008C;
+        }
+    }
+}
+
+void func_809EFDD0(EnDns* this, GlobalContext* globalCtx) {
+    if (this->actor.params == 0x9) {
+        if (CUR_UPG_VALUE(UPG_STICKS) < 2) {
+            func_8002F434(&this->actor, globalCtx, GI_STICK_UPGRADE_20, 130.0f, 100.0f);
+            return;
+        }
+        func_8002F434(&this->actor, globalCtx, GI_STICK_UPGRADE_30, 130.0f, 100.0f);
+        return;
+    }
+    if (this->actor.params == 0xA) {
+        if (CUR_UPG_VALUE(UPG_NUTS) < 2) {
+            func_8002F434(&this->actor, globalCtx, GI_NUT_UPGRADE_30, 130.0f, 100.0f);
+            return;
+        }
+        func_8002F434(&this->actor, globalCtx, GI_NUT_UPGRADE_40, 130.0f, 100.0f);
+        return;
+    }
+    func_8002F434(&this->actor, globalCtx, this->dnsItemEntry->getItemID, 130.0f, 100.0f);
+}
+
+void func_809EFEE8(EnDns* this, GlobalContext* globalCtx) {
+    if ((func_8010BDBC(&globalCtx->msgCtx) == 5) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        func_809EFDD0(this, globalCtx);
+        this->actionFunc = func_809EFF50;
+    }
+}
+
+void func_809EFF50(EnDns* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx) != 0) {
+        this->actor.parent = NULL;
+        this->actionFunc = func_809EFF98;
+        return;
+    }
+    func_809EFDD0(this, globalCtx);
+}
+
+void func_809EFF98(EnDns* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if ((player->stateFlags1 & 0x400) != 0) {
+        if ((func_8010BDBC(&globalCtx->msgCtx) == 6) && (func_80106BC8(globalCtx) != 0)) {
```

Comment:
can remove != 0

Reply:
For both the stateFlags1 and the func_80106BC8?

Reply:
i believe `func_80106BC8` returns a bool, but its not decompiled yet and not many existing examples treat it as one. its fine to leave for now i think

Reply:
Yeah, I've seen a few like that. I'll leave it for the time being